spec/api/user_spec.rb

```diff
@@ -325,5 +325,314 @@ def test_unicode_data(text)
 
       include_examples "unicode data"
     end
+
+    describe "GET /api/v1/users/:user_id/social_stats" do
+      before :each do
+        User.delete_all
+        Content.all.delete
+        @user1 = create_test_user 1
+        @user2 = create_test_user 2
+        @user3 = create_test_user 3
+        @user4 = create_test_user 4
+      end
+
+      def check_social_stats(response, excpected)
+        excpected.each do |key, value|
+          response[key].should == value
+        end
+      end
+
+      def make_social_stats(
+        num_threads=0, num_comments=0, num_replies=0, num_upvotes=0,
+        num_downvotes=0, num_flagged=0, num_comments_generated=0, num_thread_followers=0
```

Comment:
Ruby's default parameters are much less useful than python's default keyword parameters. Unlike python which respects parameter names, parameters in ruby (at least in 1.x) are always positional, so `make_social_stats(num_comments=1)` would be interpreted as `make_social_stats(1,0,0,0,0,0,0,0)` instead of `make_social_stats(0,1,0,0,0,0,0,0)` like you might expect. For that reason it doesn't make much sense to specify defaults here.